src/jit/lclvars.h

```diff
@@ -0,0 +1,750 @@
+
+    BYTE* lvGcLayout; // GC layout info for structs
+
+#if ASSERTION_PROP
+    BlockSet   lvRefBlks;          // Set of blocks that contain refs
+    GenTreePtr lvDefStmt;          // Pointer to the statement with the single definition
+    void       lvaDisqualifyVar(); // Call to disqualify a local variable from use in optAddCopies
+#endif
+    var_types TypeGet() const
+    {
+        return (var_types)lvType;
+    }
+    bool lvStackAligned() const
+    {
+        assert(lvIsStructField);
+        return ((lvFldOffset % sizeof(void*)) == 0);
+    }
+    bool lvNormalizeOnLoad() const
+    {
+        return varTypeIsSmall(TypeGet()) &&
+               // lvIsStructField is treated the same as the aliased local, see fgDoNormalizeOnStore.
+               (lvIsParam || lvAddrExposed || lvIsStructField);
+    }
+
+    bool lvNormalizeOnStore()
+    {
+        return varTypeIsSmall(TypeGet()) &&
+               // lvIsStructField is treated the same as the aliased local, see fgDoNormalizeOnStore.
+               !(lvIsParam || lvAddrExposed || lvIsStructField);
+    }
+
+    void lvaResetSortAgainFlag(Compiler* pComp);
+    void decRefCnts(BasicBlock::weight_t weight, Compiler* pComp, bool propagate = true);
+    void incRefCnts(BasicBlock::weight_t weight, Compiler* pComp, bool propagate = true);
+    void setPrefReg(regNumber regNum, Compiler* pComp);
+    void addPrefReg(regMaskTP regMask, Compiler* pComp);
+    bool IsFloatRegType() const
+    {
+        return isFloatRegType(lvType) || lvIsHfaRegArg();
+    }
+    var_types GetHfaType() const
+    {
+        return lvIsHfa() ? (lvHfaTypeIsFloat() ? TYP_FLOAT : TYP_DOUBLE) : TYP_UNDEF;
+    }
+    void SetHfaType(var_types type)
+    {
+        assert(varTypeIsFloating(type));
+        lvSetHfaTypeIsFloat(type == TYP_FLOAT);
+    }
+
+#ifndef LEGACY_BACKEND
+    var_types lvaArgType();
+#endif
+
+    PerSsaArray lvPerSsaData;
+
+#ifdef DEBUG
+    // Keep track of the # of SsaNames, for a bounds check.
+    unsigned lvNumSsaNames;
+#endif
+
+    // Returns the address of the per-Ssa data for the given ssaNum (which is required
+    // not to be the SsaConfig::RESERVED_SSA_NUM, which indicates that the variable is
+    // not an SSA variable).
+    LclSsaVarDsc* GetPerSsaData(unsigned ssaNum)
+    {
+        assert(ssaNum != SsaConfig::RESERVED_SSA_NUM);
+        assert(SsaConfig::RESERVED_SSA_NUM == 0);
+        unsigned zeroBased = ssaNum - SsaConfig::UNINIT_SSA_NUM;
+        assert(zeroBased < lvNumSsaNames);
+        return &lvPerSsaData.GetRef(zeroBased);
+    }
+
+#ifdef DEBUG
+public:
+    void PrintVarReg() const
+    {
+        if (isRegPairType(TypeGet()))
+        {
+            printf("%s:%s", getRegName(lvOtherReg), // hi32
+                   getRegName(lvRegNum));           // lo32
+        }
+        else
+        {
+            printf("%s", getRegName(lvRegNum));
+        }
+    }
+#endif // DEBUG
+
+}; // class LclVarDsc
+
+class LclVarTable final
+{
```

Comment:
I guess I'm ok with `operator[]` returning a `&` since then the calling code makes lvaTable "look" more like an array, e.g. 'LclVarDsc* tmp = &lvaTable[lclNum]`.

I actually like not having the `operator+` and forcing all callers to look the same (and changing all of them to do so)